src/utils/listenerHelpers.ts

```diff
@@ -87,3 +88,79 @@ export function propGetSet<Obj extends object, Key extends keyof Obj>(
 ) {
   return [() => obj[key], (v: Obj[Key]) => (obj[key] = v)] as const;
 }
+
+// override all keyboard inputs entering a MathQuill field
+// and optionally prevent them from doing their default behavior
+export function hookIntoOverrideKeystroke(
+  // field for which to override kb inputs
+  mq: MathQuillField,
+  // callback function
+  // return false to override all other events
+  fn: (key: string, evt: KeyboardEvent) => boolean | undefined,
+  // higher priority --> runs first
+  priority: number,
+  // unique key to prevent adding duplicate hooks
+  key: string
+) {
+  // @ts-expect-error adding properties to functions
+  if (!mq.__options.overrideKeystroke.isMonkeypatchedIn) {
+    const unsub = attach(
+      ...propGetSet(mq.__options, "overrideKeystroke"),
+      (key, evt) => {
+        // @ts-expect-error adding properties to functions
+        for (const h of mq.__options.overrideKeystroke.handlers as {
+          fn: (key: string, evt: KeyboardEvent) => boolean | undefined;
+          priority: number;
+        }[]) {
+          if (h.fn(key, evt) === false) return [false, undefined];
+        }
+      }
+    );
+
+    // @ts-expect-error adding properties to functions
+    mq.__options.overrideKeystroke.unsub = unsub;
```

Comment:
You can add these to the type of `overrideKeystroke` in `MathQuillFieldOptions`, just put a message saying you've added them.

I've done this to two things already and marked them as `Desmos will pass this through ignored`

Reply:
Only thing is I don't intend to always have these properties attached to `overrideKeystroke`. I guess I can mark them as optional.

Reply:
Another reason to be optional: Before you monkeypatch them in, they are not present on the object.

Reply:
I've implemented what might be a partial fix (?) for this: Rather than having a dedicated `hookIntoOverrideKeystroke` function, I've instead opted to have a generic `hookIntoFunction`. I've managed to do enough type wrangling to avoid any `// @ts-expect-error`s in it, but to me it wouldn't make much sense to add the properties it adds to every function it's used on since those properties are only really ever used within `hookIntoFunction` directly.